numpy/core/tests/test_function_base.py

```diff
@@ -407,3 +407,67 @@ def test_round_negative(self):
         y = linspace(-1, 3, num=8, dtype=int)
         t = array([-1, -1, 0, 0, 1, 1, 2, 3], dtype=int)
         assert_array_equal(y, t)
+
+    def test_datetime(self):
+        t1 = datetime64("2018-12-12 06:00:00")
+        t2 = datetime64("2018-12-12 07:00:00")
+        quarters = array([0, 15*60, 30*60, 45*60, 60*60], dtype="m8[s]")
+        assert_array_equal(linspace(t1, t2, 5), t1 + quarters)
+
+    def test_timedelta(self):
+        t1 = timedelta64(0, "s")
+        t2 = timedelta64(3600, "s")
+        quarters = array([0, 15*60, 30*60, 45*60, 60*60], dtype="m8[s]")
+        assert_array_equal(linspace(t1, t2, 5), t1 + quarters)
+
+    def test_nat(self):
+        t1 = datetime64("NaT")
+        t2 = datetime64("2020-01-01")
+        actual = linspace(t1, t2, 5)
+        expected = array([t1]*4+[t2])
+        assert_array_equal(actual, expected)
+        actual = linspace(t1, t2, 5, False)
```

Comment:
good catch. this requires conditioning `delta` on `np.isnat(delta)`.

Reply:
@seberg setting `stop=NaT` is problematic since the internal array `y` has to be `int64` to be compatible with `timedelta64`, but `NaN` doesn't cast to `int64`.